test/utils/timeUtils.js

```diff
@@ -0,0 +1,21 @@
+const { network } = require("hardhat");
+
+const advanceTime = async (time) =>
+    new Promise((resolve, reject) => {
```

Comment:
@bejaxer you can use `async/await` instead of Promise here right? (or if you do not want to go with async/await, drop the unused async?)